planning/behavior_velocity_planner/src/scene_module/intersection/manager.cpp

```diff
@@ -83,10 +83,21 @@ IntersectionModuleManager::IntersectionModuleManager(rclcpp::Node & node)
   ip.external_input_timeout = node.declare_parameter(ns + ".walkway.external_input_timeout", 1.0);
   ip.collision_start_margin_time = node.declare_parameter(ns + ".collision_start_margin_time", 5.0);
   ip.collision_end_margin_time = node.declare_parameter(ns + ".collision_end_margin_time", 2.0);
+}
+
+MergeFromPrivateModuleManager::MergeFromPrivateModuleManager(rclcpp::Node & node)
+: SceneModuleManagerInterface(node, getModuleName())
+{
+  const std::string ns(getModuleName());
   auto & mp = merge_from_private_area_param_;
   mp.stop_duration_sec =
     node.declare_parameter(ns + ".merge_from_private_area.stop_duration_sec", 1.0);
-  mp.intersection_param = intersection_param_;
+  mp.intersection_param.decel_velocity =
+    node.get_parameter("intersection.decel_velocity").as_double();
+  mp.intersection_param.detection_area_length =
```

Comment:
In this implementation, `mp` has many uninitialized variables. This will lead to bugs later.

I recommend following change.

```
  struct PlannerParam
  {
    IntersectionModule::PlannerParam intersection_param;
    double stop_duration_sec;
  };
```

->

```
  struct PlannerParam
  {
    double decel_velocity;
    double detection_area_length;
    double stop_line_margin;
    double stop_duration_sec;
  };
```

Reply:
@tkimura4 it's a good point but some intersection utils depends on intersection param class type so also need to change functions objective lanelet and stop line generation

Reply:
```
bool generateStopLine(
  const int lane_id, const std::vector<lanelet::CompoundPolygon3d> detection_areas,
  const std::shared_ptr<const PlannerData> & planner_data,
  const IntersectionModule::PlannerParam & planner_param,
  autoware_auto_planning_msgs::msg::PathWithLaneId * original_path,
  const autoware_auto_planning_msgs::msg::PathWithLaneId & target_path, int * stop_line_idx,
  int * pass_judge_line_idx, int * first_idx_inside_lane, const rclcpp::Logger logger)
  
  bool getObjectiveLanelets(
  lanelet::LaneletMapConstPtr lanelet_map_ptr, lanelet::routing::RoutingGraphPtr routing_graph_ptr,
  const int lane_id, const IntersectionModule::PlannerParam & planner_param,
  std::vector<lanelet::ConstLanelets> * conflicting_lanelets_result,
  std::vector<lanelet::ConstLanelets> * objective_lanelets_result, const rclcpp::Logger logger)
```

or we can get all params declared in intersection if it's not used?

Reply:
@tkimura4 
[e2b6420](https://github.com/autowarefoundation/autoware.universe/pull/447/commits/e2b642089d82db32e0c6695508aa5cf2f9502764)
includes this change and still the same behavior
![image](https://user-images.githubusercontent.com/65527974/155946824-68a23387-4d74-4a46-a028-35d7acfc8b4e.png)